controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1527,10 +1524,59 @@ bool master_thread::handle_intel_slave_join(Socket *sd, ieee1905_1::CmduMessageR
     return true;
 }
 
-bool master_thread::handle_non_intel_slave_join(Socket *sd,
-                                                std::shared_ptr<ieee1905_1::tlvWscM1> tlvwscM1,
-                                                std::string bridge_mac, std::string radio_mac,
-                                                ieee1905_1::CmduMessageTx &cmdu_tx)
+/**
+ * @brief parse the radio basic capabilities TLV and store the operating class
+ * in the database as supported channels.
+ * 
+ * @param radio_mac radio mac address
+ * @param radio_caps radio basic capabilities TLV received from the remote agent
+ * @return true on success
+ * @return false on failure
+ */
+bool master_thread::autoconfig_wsc_parse_radio_caps(
+    std::string radio_mac, std::shared_ptr<wfa_map::tlvApRadioBasicCapabilities> radio_caps)
+{
+    // read all operating class list
+    auto operating_classes_list_length = radio_caps->operating_classes_info_list_length();
+    if (operating_classes_list_length > beerocks::message::SUPPORTED_CHANNELS_LENGTH) {
+        LOG(WARNING) << "operating class info list larger then maximum supported channels";
+        operating_classes_list_length = beerocks::message::SUPPORTED_CHANNELS_LENGTH;
+    }
+    for (int oc_idx = 0; oc_idx < operating_classes_list_length; oc_idx++) {
+        std::stringstream ss;
+        auto operating_class_tuple = radio_caps->operating_classes_info_list(oc_idx);
+        if (!std::get<0>(operating_class_tuple)) {
+            LOG(ERROR) << "getting operating class entry has failed!";
+            return false;
+        }
+        auto &op_class                  = std::get<1>(operating_class_tuple);
+        auto operating_class            = op_class.operating_class();
+        auto maximum_transmit_power_dbm = op_class.maximum_transmit_power_dbm();
+        ss << "operating_class=" << int(operating_class);
+        ss << "maximum_transmit_power_dbm=" << int(maximum_transmit_power_dbm);
+        ss << ", statically_non_operable_channel_list={";
+
+        auto non_oper_channels_list_length =
+            op_class.statically_non_operable_channels_list_length();
+        for (int ch_idx = 0; ch_idx < non_oper_channels_list_length; ch_idx++) {
+            auto ch_tuple   = op_class.statically_non_operable_channels_list(ch_idx);
+            auto channel_rx = std::get<1>(ch_tuple);
+            ss << int(channel_rx);
+
+            // add comma if not last channel in the list, else close list by add curl brackets
+            ss << (((ch_idx + 1) != non_oper_channels_list_length) ? "," : "}");
+        }
+        LOG(DEBUG) << ss.str();
```

Comment:
Is it really worth adding all this code for a single debug message?

Reply:
Now it is, that I changed the API to receive statically non operable channels.